src/Avalonia.Controls/DateTimePickers/DatePicker.cs

```diff
@@ -103,134 +93,142 @@ public class DatePicker : TemplatedControl
+
         public string DayFormat
         {
-            get => _dayFormat;
-            set => SetAndRaise(DayFormatProperty, ref _dayFormat, value);
+            get => GetValue(DayFormatProperty);
+            set => SetValue(DayFormatProperty, value);
         }
 
         /// <summary>
         /// Gets or sets whether the day is visible
         /// </summary>
         public bool DayVisible
         {
-            get => _dayVisible;
-            set
-            {
-                SetAndRaise(DayVisibleProperty, ref _dayVisible, value);
-                SetGrid();
-            }
+            get => GetValue(DayVisibleProperty);
+            set => SetValue(DayVisibleProperty, value);
         }
 
         /// <summary>
         /// Gets or sets the maximum year for the picker
         /// </summary>
         public DateTimeOffset MaxYear
         {
-            get => _maxYear;
-            set
-            {
-                if (value < MinYear)
-                    throw new InvalidOperationException("MaxDate cannot be less than MinDate");
-                SetAndRaise(MaxYearProperty, ref _maxYear, value);
+            get => GetValue(MaxYearProperty);
+            set => SetValue(MaxYearProperty, value);
+        }
 
-                if (SelectedDate.HasValue && SelectedDate.Value > value)
-                    SelectedDate = value;
+        private static DateTimeOffset CoerceMaxYear(AvaloniaObject sender, DateTimeOffset value)
+        {
+            if (value < sender.GetValue(MinYearProperty))
+            {
+                throw new InvalidOperationException($"{MaxYearProperty.Name} cannot be less than {MinYearProperty.Name}");
             }
+
+            return value;
+        }
+
+        private void OnMaxYearChanged(DateTimeOffset? value)
+        {
+            if (SelectedDate.HasValue && SelectedDate.Value > value)
+                SetCurrentValue(SelectedDateProperty, value);
         }
 
         /// <summary>
         /// Gets or sets the minimum year for the picker
         /// </summary>
         public DateTimeOffset MinYear
         {
-            get => _minYear;
-            set
-            {
-                if (value > MaxYear)
-                    throw new InvalidOperationException("MinDate cannot be greater than MaxDate");
-                SetAndRaise(MinYearProperty, ref _minYear, value);
+            get => GetValue(MinYearProperty);
+            set => SetValue(MinYearProperty, value);
+        }
 
-                if (SelectedDate.HasValue && SelectedDate.Value < value)
-                    SelectedDate = value;
+        private static DateTimeOffset CoerceMinYear(AvaloniaObject sender, DateTimeOffset value)
+        {
+            if (value > sender.GetValue(MaxYearProperty))
+            {
+                throw new InvalidOperationException($"{MinYearProperty.Name} cannot be greater than {MaxYearProperty.Name}");
```

Comment:
I wonder if there will be the same behavior. Does it crash the app or shows the error in data validation error control?

Reply:
I investigated and there has been no change in behaviour. The exception has always propagated, potentially causing a crash. Crashes in coercion methods also propagate in WPF.

I'm not sure about the wisdom of throwing an exception in this case, but that's what it does.